Update CONTRIBUTING.md w/ detail on opening issues

benvinegar opened the pull request:
Again, borrowed heavily from the [Redux project](https://github.com/reactjs/redux/blob/master/CONTRIBUTING.md).

cc @mattrobenolt @LewisJEllis

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/getsentry/raven-js/726)
<!-- Reviewable:end -->

LewisJEllis commented:
Review status: 0 of 1 files reviewed at latest revision, 2 unresolved discussions.

---

*[CONTRIBUTING.md, line 27 \[r1\]](https://reviewable.io:443/reviews/getsentry/raven-js/726#-KREw1Y46NRXLbqHZyhy:-KREw1Y50VqrVJICtTUV:b6nu6pi) ([raw file](https://github.com/getsentry/raven-js/blob/68081382f6a70db8f4ee59aa329a02a9a610c573/CONTRIBUTING.md#L27)):*
> ```Markdown
> * If you are using the Raven CDN (http://ravenjs.com)
> 
> Lastly, it is strongly encouraged to provide a small project reproducing your issue. You can put your code on JSBin or, for bigger projects, on GitHub. Make sure all the necessary dependencies are declared in package.json so anyone can run npm install && npm start and reproduce your issue.
> ```

We recommend JSBin here but JSFiddle in the template; let's either pick one or list both in both places?

---

*[CONTRIBUTING.md, line 112 \[r1\]](https://reviewable.io:443/reviews/getsentry/raven-js/726#-KREx96eSzmflOreARXx:-KREx96eSzmflOreARXy:bg9cpg0) ([raw file](https://github.com/getsentry/raven-js/blob/68081382f6a70db8f4ee59aa329a02a9a610c573/CONTRIBUTING.md#L112)):*
> ```Markdown
> _This is a checklist for core contributors when releasing a new version._
> 
>   * [ ] Verify TypeScript [language definition file](https://github.com/getsentry/  * raven-js/blob/master/typescript/raven.d.ts) is up to date
> ```

Maybe this is some markdown-fu that I don't know, but what are all these near-end-of-line asterisks for? Looks like maybe a regex gone wild

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/getsentry/raven-js/726)*
<!-- Sent from Reviewable.io -->

benvinegar commented:
Review status: 0 of 1 files reviewed at latest revision, 2 unresolved discussions.

---

*[CONTRIBUTING.md, line 27 \[r1\]](https://reviewable.io:443/reviews/getsentry/raven-js/726#-KREw1Y46NRXLbqHZyhy:-KREyx6Vi5AtyGt8FtxR:btbtigc) ([raw file](https://github.com/getsentry/raven-js/blob/68081382f6a70db8f4ee59aa329a02a9a610c573/CONTRIBUTING.md#L27)):*
<details><summary><i>Previously, LewisJEllis (Lewis J Ellis) wrote…</i></summary>
> We recommend JSBin here but JSFiddle in the template; let's either pick one or list both in both places?
</details>
Switched to JSFiddle.

---

*[CONTRIBUTING.md, line 112 \[r1\]](https://reviewable.io:443/reviews/getsentry/raven-js/726#-KREx96eSzmflOreARXx:-KREz0LsxhTbTI6OQ8sq:b-j0xkk8) ([raw file](https://github.com/getsentry/raven-js/blob/68081382f6a70db8f4ee59aa329a02a9a610c573/CONTRIBUTING.md#L112)):*
<details><summary><i>Previously, LewisJEllis (Lewis J Ellis) wrote…</i></summary>
> Maybe this is some markdown-fu that I don't know, but what are all these near-end-of-line asterisks for? Looks like maybe a regex gone wild
</details>
This was me not paying attention to line wrapping in my editor. Fixed.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/getsentry/raven-js/726)*
<!-- Sent from Reviewable.io -->

LewisJEllis commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Reviewed 1 of 1 files at r1, 1 of 1 files at r2.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/getsentry/raven-js/726#-:-KREzwjKLLrMy6UAs7AS:bnfp4nl)*
<!-- Sent from Reviewable.io -->
